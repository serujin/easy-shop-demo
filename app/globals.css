@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
