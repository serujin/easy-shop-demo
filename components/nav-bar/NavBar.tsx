import React from "react";
import NavigationItem from "./navigation-item/NavigationItem";
import NavBarLogo from "./logo/NavBarLogo";
import styles from "./NavBar.module.css";

const NAVIGATION_LIST_ITEMS = ["Products", "Tools", "Samples", "Deals"];
const NAVIGATION_USER_ITEMS = ["Log in", "Sign up"];

const renderItems = (list: string[]) => (
  <ul className={styles.navigationList}>
    {list.map((text, index) => (
      <NavigationItem key={index} text={text} />
    ))}
  </ul>
);

export default function NavBar() {
  return (
    <div className={styles.navBar}>
      <div className={styles.navBarLeft}>
        <NavBarLogo variant="horse" />
        <nav className={styles.navigation}>
          {renderItems(NAVIGATION_LIST_ITEMS)}
        </nav>
      </div>
      <div className={styles.navBarRight}>
        <NavBarLogo variant="cart" />
        {renderItems(NAVIGATION_USER_ITEMS)}
      </div>
    </div>
  );
}
