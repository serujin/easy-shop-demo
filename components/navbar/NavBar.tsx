import Image from "next/image";
import React from "react";
import Link from "next/link";
import styles from "./NavBar.module.css";
import NavigationItem from "./NavigationItem";

const NAVIGATION_LIST_ITEMS = ["Products", "Tools", "Samples", "Deals"];
const NAVIGATION_USER_ITEMS = ["Log in", "Sign up"];

export default function NavBar() {
  return (
    <div className={styles.navBar}>
      <div className={styles.navBarLeft}>
        <Link href={"/"}>
          <Image
            alt={""}
            className={styles.logoLeft}
            src={"/navBar/horse-logo.svg"}
            width={22}
            height={22}
          />
        </Link>
        <nav className={styles.navigation}>
          <ul className={styles.navigationList}>
            {NAVIGATION_LIST_ITEMS.map((text, index) => (
              <NavigationItem key={index} text={text} />
            ))}
          </ul>
        </nav>
      </div>
      <div className={styles.navBarRight}>
        <Link href={"/"}>
          <Image
            alt={""}
            className={styles.logoRight}
            src={"/navBar/cart.svg"}
            width={20}
            height={20}
          />
        </Link>
        <ul className={styles.navigationList}>
          {NAVIGATION_USER_ITEMS.map((text, index) => (
            <NavigationItem key={index} text={text} />
          ))}
        </ul>
      </div>
    </div>
  );
}
