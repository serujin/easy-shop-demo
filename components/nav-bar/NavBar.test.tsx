import React from "react";
import { render } from "@testing-library/react";
import NavBar from "./NavBar";
import { describe, expect, it } from "@jest/globals";

describe("NavBar component", () => {
  it("should render correctly", () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
