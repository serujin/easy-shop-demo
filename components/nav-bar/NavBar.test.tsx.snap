// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render correctly 1`] = `
<div>
  <div
    class="navBar"
  >
    <div
      class="navBarLeft"
    >
      <a
        href="/"
      >
        <img
          alt="Horse logo"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/navBar/horse.svg"
          style="color: transparent;"
          width="20"
        />
      </a>
      <nav
        class="navigation"
      >
        <ul
          class="navigationList"
        >
          <li>
            <a
              class="navigationLink"
              href="#"
            >
              <b>
                Products
              </b>
            </a>
          </li>
          <li>
            <a
              class="navigationLink"
              href="#"
            >
              <b>
                Tools
              </b>
            </a>
          </li>
          <li>
            <a
              class="navigationLink"
              href="#"
            >
              <b>
                Samples
              </b>
            </a>
          </li>
          <li>
            <a
              class="navigationLink"
              href="#"
            >
              <b>
                Deals
              </b>
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="navBarRight"
    >
      <a
        href="/"
      >
        <img
          alt="Cart logo"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/navBar/cart.svg"
          style="color: transparent;"
          width="20"
        />
      </a>
      <ul
        class="navigationList"
      >
        <li>
          <a
            class="navigationLink"
            href="#"
          >
            <b>
              Log in
            </b>
          </a>
        </li>
        <li>
          <a
            class="navigationLink"
            href="#"
          >
            <b>
              Sign up
            </b>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
